pkg/webui/lib/components/with-locale.js

```diff
@@ -12,193 +12,179 @@
+          import(
+            /* webpackChunkName: "lang.[request]" */ `../../locales/.backend/${language}.json`
+          ),
+        ]
+      } else {
+        promises = [
+          import(/* webpackChunkName: "lang.[request]" */ `../../locales/${language}.json`),
+        ]
+      }
 
-    // Load the language files if needed.
-    await this.setState({ loaded: false })
-
-    let promises = []
-    let withLocale = false
-    if (language !== defaultLanguage) {
-      withLocale = true
-      promises = [
-        import(/* WebpackChunkName: "lang.[request]" */ `../../locales/${language}.json`), // Frontend messages
-        import(/* WebpackChunkName: "lang.[request]" */ `../../locales/.backend/${language}.json`), // Backend messages
-      ]
-    }
+      if (!window.Intl.NumberFormat || !window.Intl.DateTimeFormat) {
+        log(`Polyfilling locale ${locale} for language ${language}`)
+        promises.push(import('intl'))
+        promises.push(
+          import(/* webpackChunkName: "locale.[request]" */ `intl/locale-data/jsonp/${locale}`),
+        )
+      }
 
-    if (!window.Intl.NumberFormat || !window.Intl.DateTimeFormat) {
-      log(`Polyfilling locale ${locale} for language ${language}`)
-      promises.push(import('intl'))
-      promises.push(
-        import(/* WebpackChunkName: "locale.[request]" */ `intl/locale-data/jsonp/${locale}`),
-      )
-    }
+      if (!window.Intl.PluralRules) {
+        log(`Polyfilling Intl.PluralRules`)
+        promises.push(/* webpackChunkName: "locale-plural-rules" */ import('intl-pluralrules'))
+      }
 
-    if (!window.Intl.PluralRules) {
-      log(`Polyfilling Intl.PluralRules`)
-      promises.push(import('intl-pluralrules'))
-    }
+      if (!window.Intl.RelativeTimeFormat) {
+        log(`Polyfilling Intl.RelativeTimeFormat data for language ${language}`)
+        promises.push(
+          import(
+            /* webpackChunkName: "locale-time-polyfill" */ '@formatjs/intl-relativetimeformat/polyfill'
+          ),
+        )
+        promises.push(
+          import(
+            /* webpackChunkName: "locale-time-locales.[request]" */ `@formatjs/intl-relativetimeformat/locale-data/${language}`
+          ),
+        )
+      }
 
-    if (!window.Intl.RelativeTimeFormat) {
-      log(`Polyfilling Intl.RelativeTimeFormat data for language ${language}`)
-      promises.push(import('@formatjs/intl-relativetimeformat/polyfill'))
-      promises.push(import(`@formatjs/intl-relativetimeformat/locale-data/${language}`))
+      promise.current = CancelablePromise.resolve(Promise.all(promises))
+      try {
+        const res = await promise.current
+        setLocale(locale)
+        if (language === defaultLanguage) {
+          // Merge frontend and backend messages.
+          setMessages({ ...res[0], ...res[1] })
+        } else {
+          setMessages(res[0])
+        }
+        localStorage.setItem('lang', language)
+      } catch (err) {
+        handleIntlError(err)
```

Comment:
If something goes wrong in `try` section, the Console hangs with the spinner. I think we should display an error or proceed with default locale.